### DISCUSSION TITLE
Is the RHVAE in pythae same as the RHVAE in A Geometric Perspective on Variational Autoencoders?

### DISCUSSION THREAD

**Discussion by axu-git** [clementchadebec/benchmark_VAE#127] [General]
There are two implementations of RHVAEs. The one in pythae refers to the model introduced in the paper "Data Augmentation in High Dimensional ..." There is another one (https://github.com/clementchadebec/geometric_perspective_on_vaes) that refers to the model used in "Geometry-Aware Hamiltonian Variational Auto-Encoder". Are they the same model?

I would like to try the RHVAE in Geometry-Aware paper, and I am wondering if I can use the one implemented in pythae or if I need to use the one in the github for that paper (https://github.com/clementchadebec/geometric_perspective_on_vaes).

**Comment by clementchadebec** [comment]
Hi @axu-git,

These two implementations are indeed similar. I recommend using the one in `pythae` since it allows accessing to other features such as easily switching encoder/decoder architectures, quickly compute latents etc...

Best,

Clément

**Reply by axu-git** [reply]
Thank you!! Just to clarify, the models are the same (i.e., there would be no reason to use the one in the Geometry-Aware github over the one in pythae)?

**Reply by clementchadebec** [reply]
Yep, that is correct. The models are the same. Just make sure to set the encoder and decoder in Pythae using the below architectures if you want to reproduce the results in 
https://github.com/clementchadebec/geometric_perspective_on_vaes

https://github.com/clementchadebec/geometric_perspective_on_vaes/blob/46c95740fff3bbe4eb116e119d3782c2083f8730/models/vae.py#L25

**Reply by axu-git** [reply]
Thank you!!
